luarules/gadgets/game_team_power_watcher.lua

```diff
@@ -62,233 +29,462 @@ if gadgetHandler:IsSyncedCode() then
+    GG.PowerLib.UnitsWithPower[unitID] = {unitID = unitID, power = UnitDefs[unitDefID].power, team = unitTeam}
+    GG.PowerLib.TeamPowers[unitTeam] = (GG.PowerLib.TeamPowers[unitTeam] + UnitDefs[unitDefID].power) or UnitDefs[unitDefID].power
+    
+    --debug
 
-            --update peak powers
-            if teamPowers[unitTeam] and peakTeamPowers[unitTeam] < teamPowers[unitTeam] then
-                peakTeamPowers[unitTeam] = teamPowers[unitTeam]
-            end
+    --testPowerTable = HighestAlliedPeakPower(unitTeam)
+    --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, GG.PowerLib.TeamPowers[unitTeam], "test result", testPowerTable.teamID, testPowerTable.power)
 
+    --testPowerNumber = TechGuesstimate(GG.PowerLib.TeamPowers[unitTeam])
+    --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, GG.PowerLib.TeamPowers[unitTeam], "test result", testPowerNumber)
 
-            --Debug row, eventually to be replaced with %frame triggered calculation events
-            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "highest", highestTeamPower.teamID, highestTeamPower.power)
-            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageTeamPower", averageTeamPower)
-            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageAlliedTeamPower", averageAlliedTeamPower)
-            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageTechGuesstimate", averageTechGuesstimate)
-            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageHumanTechGuesstimate", averageHumanTechGuesstimate)
-            Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "averageAlliedTechGuesstimate", averageAlliedTechGuesstimate)
-            --Spring.Echo(UnitDefs[unitDefID].name, unitTeam, UnitDefs[unitDefID].power, teamPowers[unitTeam], "peakTeamPowers", peakTeamPowers[unitTeam])
-		end
-	end
-
-
-
-	function gadget:UnitDestroyed(unitID, unitDefID, unitTeam)
-        if unitsWithPower[unitID] then
-			unitsWithPower[unitID] = nil
-		end
-        if UnitDefs[unitDefID].power then
-            if teamPowers[unitTeam] then
-                if teamPowers[unitTeam] <= UnitDefs[unitDefID].power then
-                    teamPowers[unitTeam] = 0
-                else
-                    teamPowers[unitTeam] = teamPowers[unitTeam] - UnitDefs[unitDefID].power
-                end
+    --update peak powers
+    if GG.PowerLib.TeamPowers[unitTeam] and GG.PowerLib.PeakTeamPowers[unitTeam] < GG.PowerLib.TeamPowers[unitTeam] then
+        GG.PowerLib.PeakTeamPowers[unitTeam] = GG.PowerLib.TeamPowers[unitTeam]
+    end
+end
+
+function gadget:UnitDestroyed(unitID, unitDefID, unitTeam)
```

Comment:
Does this also need to recalculate peak team powers? Or does peak power measure the historical maximum?

Reply:
Historical maximum. The highest ever achieved.

Reply:
This is used for example in the scavenger gadget to capture largest power of the last wave (since they're always getting bigger)

And also will be useful for dynamic evocom evolution condition where once a certain threshold is reached you don't want it to ever go down.